CMakeLists.txt

```diff
@@ -10,7 +16,29 @@ include(GNUInstallDirs)
 include(BuildPlatformDetection)
 include(SetProjectDependencies)
 
-set(PYTHON_EXECUTABLE ${CMAKE_INSTALL_PREFIX}/bin/python3)
+# For all OSs:
+if($ENV{PYTHONPATH})
+        set(ENV{PYTHONPATH} ${CMAKE_INSTALL_PREFIX}/lib/python3/dist-packages/:${CMAKE_INSTALL_PREFIX}/lib/python3.5:${CMAKE_INSTALL_PREFIX}/lib/python3.5/site-packages:$ENV{PYTHONPATH})
+    else()
+        set(ENV{PYTHONPATH} ${CMAKE_INSTALL_PREFIX}/lib/python3/dist-packages/:${CMAKE_INSTALL_PREFIX}/lib/python3.5:${CMAKE_INSTALL_PREFIX}/lib/python3.5/site-packages)
+    endif()
+
+# OS-specific changes
+if(BUILD_OS_LINUX)
+    # Preparing for building into a prefix on Linux
+    set(ENV{PATH} ${CMAKE_INSTALL_PREFIX}/bin/:$ENV{PATH} CACHE STRING "PATH" FORCE)
+    set(ENV{LD_LIBRARY_PATH} ${CMAKE_INSTALL_PREFIX}/lib/)
+    set(PYTHON_EXECUTABLE python3)
+    set(PYTHON_EXECUTABLE_PREFIXED LD_LIBRARY_PATH=$ENV{LD_LIBRARY_PATH} PATH=$ENV{PATH} PYTHONPATH=$ENV{PYTHONPATH} ${PYTHON_EXECUTABLE})
```

Comment:
There should be no need for this with the changes to RPATH I did. PYTHONPATH does not need to be set since Python deduces the path correctly.